### ISSUE TITLE
[md-icon] should work the same for custom and built-in icons

### ISSUE THREAD

**Issue by ross-nordstrom** [angular/material2#1714]
#### Bug, feature request, or proposal:
MdIcon should encapsulate the fact that you're using a custom SVG icon, so the usage stays the same whether you're using built-in icons or custom ones.

#### What is the expected behavior?
I would like to reference icons the same, whether they are built-in or custom icons I've registered.

```html
<md-icon>search</md-icon>
<md-icon>my_custom_icon</md-icon>
```

If I've overridden `search` with a custom icon, I want the html usage above to remain the same.
MdIcon will use the custom icon if I've overridden a built-in icon name.

```js
mdIconRegistry.addSvgIcon('search', '/icons/my-custom-search-icon.svg')
mdIconRegistry.addSvgIcon('my_custom_icon', '/icons/my-custom-search-icon.svg')
```

#### What is the current behavior?
Consumers of your custom icons need to know they're custom and access them via a special input

```html
<!-- Built-in -->
<md-icon>search</md-icon>

<!-- Must be changed when you add a custom icon -->
<md-icon svgIcon="my_custom_icon"></md-icon>
```

#### What are the steps to reproduce?
Register custom icons and try to use them like a normal icon.

http://embed.plnkr.co/4Sz8JuTscrj4XoO7X4LA/

Plunkr demonstrates
1. built-in icons 
2. custom icons  
3. how I'd like to use custom icons

#### What is the use-case or motivation for changing an existing behavior?
I want to be able to dynamically define what Icons are used in my product. 

In some cases, I've added custom icons that aren't present in Material's Icons.
I want to be able to reference these the same as the built-in icons. This will keep
my code simple and naive, not needing to care what "kind" of icons it's being told to render.

In other words, this change would let me override and extend my Icon set **without changing any of my Icon-consuming code.**

#### Which versions of Angular, Material, OS, browsers are affected?
Angular = `2.1.0`
Material = `2.0.0-alpha.9-3`

#### Is there anything else we should know?
This seems like an fairly easy change to implement:

```js
const providedValue = ...transcluded icon string... // <md-icon>the_icon_in_question</md-icon>

if(isCustomIcon(providedValue)) {
    useCustomIcon(providedValue); // TODO: implement
} else {
    useBuiltinIcon(providedValue); // TODO: implement
}
```

```js
isCustomIcon(providedValue) {
    // Is the icon name registered? (using '!!' to cast to boolean)
    return !!mdIconRegistry._svgIconConfigs.get(`:${providedValue}`);
}
```

**Comment by jelbourn** [comment]
This isn't "custom" vs. "built-in"- the difference in APIs is font-icons vs. SVG icons. Font-icons literally _are_ the text content of the element because they use ligatures, whereas there is no corresponding concept for SVG.

**Comment by s3ppo** [comment]
<md-icon svgSrc="/assets/sun.svg"></md-icon> is not working since beta. 
anybody can verify that?

thx and LG
Harald

**Comment by da45** [comment]
@s3ppo yes , it is not warking !!

**Comment by jelbourn** [comment]
`svgSrc` has been removed. Please see the changelog: https://github.com/angular/material2/blob/master/CHANGELOG.md

**Comment by s3ppo** [comment]
where can i find documents on how to use this sanitize? 
btw. the svgSrc is still in the icon README
